client/src/flux/stores/MessageStore.js

```diff
@@ -0,0 +1,155 @@
+import { EventEmitter } from 'events';
+import ImmutableMap from '../models/groupList';
+import Dispatcher from '../Dispatcher';
+import ActionTypes from '../ActionTypes';
+
+const CHANGE_EVENT = 'change';
+
+// create empty immutable map to hold message list
+let messages = new ImmutableMap();
+
+const notificationMap = new Map();
+
+let groupWithNewMessageId;
+
+// add new messages to map of messages
+const addNewMessageGroup = (newMessageGroup) => {
+  messages = messages.merge(newMessageGroup);
+};
+
+
+/**
+ * MessageStore holds store logic for messages
+ * @return {void}
+ */
+class MessageStore extends EventEmitter {
+  /**
+   * addChangeListener
+   * @memberof MessageStore
+   * @param {Function} callback
+   * 
+   * @param {String} CHANGE_EVENT_ID
+   * 
+   * @return {void}
+   */
+  addChangeListener(callback, CHANGE_EVENT_ID) {
+    this.on(CHANGE_EVENT_ID, callback);
+  }
+
+  /**
+  * removeChangeListener
+  * @memberof MessageStore
+  *
+  * @param {Function} callback
+  * @param {String} CHANGE_EVENT_ID
+  *
+  * @return {void}
+  */
+  removeChangeListener(callback, CHANGE_EVENT_ID) {
+    this.removeListener(CHANGE_EVENT_ID, callback);
+  }
+
+  /** add listener for notification changes
+   * @param {Function} callback
+   * 
+   * @return {void}
+   */
+  addNotificationChangeListener(callback) {
+    this.on(CHANGE_EVENT, callback);
+  }
+
+  /** remove notification change event listener
+   * @param {Function} callback
+   * 
+   * @return {void}
+   */
+  removeNotificationChangeListener(callback) {
+    this.removeListener(CHANGE_EVENT, callback);
+  }
+
+  /* eslint class-methods-use-this: 0 */
+
+  /**
+   * GetMessage
+   * @memberof MessageStore
+   * 
+   * @param {String} id
+   * 
+   * @return {Map} messages.get(id)
+   */
+  getMessage(id) {
+    return messages.get(id);
+  }
+
+  /**
+   * Get the notification properties of the groups
+   * @return {Object} groupNotificationDetatls
+   */
+  getGroupNotificationDetails() {
+    return {
+      Id: groupWithNewMessageId,
+      status: notificationMap
+    };
+  }
+}
+
+const messageStore = new MessageStore();
+
+Dispatcher.register((payload) => {
+  const action = payload.action;
+  let groupId;
```

Comment:
do you need all these variables defined here.
Can't you define in the message store constructor